core/modules/system/system.module

```diff
@@ -4211,12 +4211,12 @@ function path_autocomplete($string = '') {
 
     // Get taxonomy term matches.
     if (module_exists('taxonomy') && $match_count < $range) {
-      $results = db_select('taxonomy_term_data')
-        ->fields('taxonomy_term_data', array('tid', 'vocabulary', 'name'))
-        ->condition('name', '%' . db_like($string) . '%', 'LIKE')
+      $results = db_select('taxonomy_term_data', 'ttd')
```

Comment:
Instead of introducing a new alias, I suggest using the default provided by db_select
```suggestion
      $results = db_select('taxonomy_term_data')
```